Rework plugins doc and add the mapper-size plugin (#7646)
opensearch-project/documentation-website: _install-and-configure/plugins.md

layout: default
title: Installing plugins
nav_order: 90
has_children: true
redirect_from:
   - /opensearch/install/plugins/
   - /install-and-configure/install-opensearch/plugins/
---

# Installing plugins

OpenSearch is comprised of a number of plugins that add features and capabilities to the core platform. The plugins that are available to you are dependent on how OpenSearch was installed and which plugins were subsequently added or removed. For example, the minimal distribution option of OpenSearch enables only core functionality, such as indexing and search. Using the minimal distribution of OpenSearch is beneficial when working in a testing environment, you have custom plugins, or are intending to integrate OpenSearch with other services.

The standard distribution of OpenSearch has a lot more functionality bundled in. You can choose to add additional plugins or remove any of the plugins you don't need. 

> For a sampling of the available plugins, see [Available plugins](#available-plugins).

Reviewer: @natebower - While the majority are there, it's not a complete list. Do you have a better way to say this?

> For a sampling of the available plugins, see [Available plugins](#available-plugins).

Reviewer 2: @hdhalter I think the best solution would be to complete the list, but if we can't do that, then "For a list of some of the available plugins" will work.

> For a sampling of the available plugins, see [Available plugins](#available-plugins).

Author: Its not possible to list all plugins in existence. Anyone is free to create a plugin.

There are some plugins that are shipped with OpenSearch that are not installed by default and that is what this list contains. These are included in the distribution, but still need to be explicitly installed in order for a cluster to gain the functionality provided by the plugins in this list.

If there are popular community plugins then I think we could consider giving them real estate and exposure on the documentation website, but its not possible to list all of the plugins in existence.

> For a sampling of the available plugins, see [Available plugins](#available-plugins).

Reviewer 2: Then "For a list of some of the available plugins" will work.

> For a plugin to work properly with OpenSearch, it must have the ability to access data and metadata in the cluster. Consequently, to safeguard your cluster's data integrity, it is crucial that you understand a plugin's functionality before installation. Additionally, when opting for a community-provided plugin, ensure that the source is trustworthy and reliable.

Author: wdyt? `For a plugin to work properly with OpenSearch, it may request certain permissions as part of the installation process. Please review the requested permissions and proceed accordingly.`

Reviewer: Sounds good!

### Additional plugins

> There are many more plugins available on top of what is provided by default. These additional plugins have been built by OpenSearch developers or members of the OpenSearch community. For a sampling of additional plugins you can install, see [Additional plugins]({{site.url}}{{site.baseurl}}/install-and-configure/additional-plugins/index/).

Reviewer 3: Use the same language as the additional plugins intro.

```suggestion
There are many more plugins available on top of what is provided in the default distribution. These additional plugins have been built by OpenSearch developers or members of the OpenSearch community. For a sampling of additional plugins you can install, see [Additional plugins]({{site.url}}{{site.baseurl}}/install-and-configure/additional-plugins/index/).
```

## Plugin compatibility

> A plugin can specify compatibility with a particular OpenSearch version by listing that version in its `plugin-descriptor.properties` file. For example, a plugin with the following property is compatible only with OpenSearch 2.3.0:

Reviewer: ```suggestion
You can specify plugin compatibility with a particular OpenSearch version in the `plugin-descriptor.properties` file. For example, a plugin with the following property is compatible only with OpenSearch 2.3.0:
```

> A plugin can explicitly specify compatibility with a specific OpenSearch version by listing that version in its `plugin-descriptor.properties` file. For example, a plugin with the following property is compatible only with OpenSearch 2.3.0:

Author: Plugin developers specify compatibility.

> A plugin can explicitly specify compatibility with a specific OpenSearch version by listing that version in its `plugin-descriptor.properties` file. For example, a plugin with the following property is compatible only with OpenSearch 2.3.0:

Reviewer 2: @cwperks I like it!

```properties
opensearch.version=2.3.0
```

> Alternatively, a plugin can specify a range of compatible OpenSearch versions by setting the `dependencies` property in its `plugin-descriptor.properties` file using one of the following notations:

Reviewer: ```suggestion
Alternatively, you can specify a range of compatible OpenSearch versions by setting the `dependencies` property in the `plugin-descriptor.properties` file to one of the following notations:
```
